snf-image-helper/tasks/35InitializeDatasource.in

```diff
@@ -0,0 +1,93 @@
+#! /bin/bash
+
+# Copyright (C) 2017 GRNET S.A.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+### BEGIN TASK INFO
+# Provides:		InitializeDatasource
+# RunBefore:		EnforcePersonality
+# RunAfter:		MountImage
+# Short-Description:	Initialized the cloud-init NoCloud datasource
+### END TASK INFO
+
+set -e
+. "@commondir@/common.sh"
+
+initialize_task mounted_excludable overwritable
+
+if ! check_yes_no SNF_IMAGE_PROPERTY_CLOUD_INIT; then
+    warn "Cloud-init not enabled for this instance"
+    return 0
+fi
+
+CFG=$(get_cloud_init_config "$SNF_IMAGE_TARGET" "$CLOUD_INIT_CONFNAME")
+
+cat >> $CFG <<EOF
+datasource_list: [ NoCloud ]
+EOF
+
+if [ -n "$SNF_IMAGE_CLOUD_INIT_DEBUG" ]; then
+    cat >> "$CFG" <<-EOF
+	logcfg: |
+	  [loggers]
+	  keys=root,cloudinit
+	  [handlers]
+	  keys=ch,cf
+	  [formatters]
+	  keys=
+	  [logger_root]
+	  level=DEBUG
+	  handlers=
+	  [logger_cloudinit]
+	  level=DEBUG
+	  qualname=cloudinit
+	  handlers=ch,cf
+	  [handler_ch]
+	  class=StreamHandler
+	  level=DEBUG
+	  args=(sys.stderr,)
+	  [handler_cf]
+	  class=FileHandler
+	  level=DEBUG
+	  args=('/tmp/cloud-init-debug.log',)
```

Comment:
I am not very fond of writing the debug logs to /tmp. I'd rather have a /var/log/snf-image directory for the logs.